server/src/test/java/org/elasticsearch/common/lucene/search/FilteredCollectorTests.java

```diff
@@ -39,7 +41,7 @@ public void setUp() throws Exception {
         super.setUp();
         directory = newDirectory();
         RandomIndexWriter writer = new RandomIndexWriter(random(), directory, newIndexWriterConfig());
-        numDocs = randomIntBetween(10, 100);
+        numDocs = randomIntBetween(900, 1000);
```

Comment:
This is to increase the likelihood of leveraging concurrency. It still does not happen super often, around 10% of the runs. Adding flushes would help but random index writer already does it and it also slows down the tests. I have opened [12369](https://github.com/apache/lucene/pull/12369) to potentially address this in Lucene,

Reply:
The creation of segments and slices should be fully reproducible with the seed, if it's now it sounds like a bug.